src/Boo.Lang.Compiler/TypeSystem/Types.cs

```diff
@@ -121,27 +122,21 @@ public static class Types
 
 		public static readonly Type CompilerGeneratedAttribute = typeof(System.Runtime.CompilerServices.CompilerGeneratedAttribute);
 
-		static Type FindType(string typename, string typenamespace)
+		static Type FindType(string fullTypeName, string[] possibleAssemblyNames)
 		{
-			foreach (System.Reflection.Assembly assembly in AppDomain.CurrentDomain.GetAssemblies())
-			{
-				foreach (Type type in assembly.GetTypes())
```

Comment:
Seems like this could be a much less invasive change just by replacing the inner loop with a checked assembly.GetType(fullTypeNameConstructedFromTypenamespaceAndTypename)